### PR TITLE
Fixes #1719 : Added Before Model for redirecting to home page

### LINKED ISSUE #1719: Accessing private endpoints should redirect to home page.
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behaviour:**

Accessing endpoints like `http://badgeyay.com/my-badges`, if one is not logged in, shows loading and then a blank page. 
**Expected behaviour:**

Instead we should redirect the user to home page.
**Steps to reproduce:**
Go to [badgeyay](http://badgeyay.com/), and logout if you are currently logged in and then go to [my-badges](http://badgeyay.com/my-badges) page.

---
_Working on it_

### PR DESCRIPTION
**Pull request by sk9331657**
Fixes #1719 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

-This will route create-badges to login page in case user is not logged in.
ezgif com-video-to-gif 4

-

### CODE REVIEW

**Comment by sk9331657** [comment]
@ParthS007  This can be merged. Actually, there were conflicts during rebase, so it took time to resolve them.

**Comment by rupav** [comment]
Atleast tell in the issue, that you are working on it.

**Comment by sk9331657** [comment]
> Atleast tell in the issue, that you are working on it.

I will make sure next time. @rupav

**Comment by sk9331657** [comment]
@gabru-md  @abishekvashok  Please merge this.

**Comment by sk9331657** [comment]
@gabru-md  Can this be merged now or shall I update this with other routes??

**Comment by sk9331657** [comment]
@ParthS007  Done Squashing Of Commit And also added routing for my-badges and settings routes in case user is not logged in .

**Comment by ParthS007** [comment]
Please change your commit message to same as PR title.

**Comment by sk9331657** [comment]
@ParthS007  Update Commit Message :+1:

**Comment by sk9331657** [comment]
@ParthS007 @gabru-md  I have updated the changes suggested by you bros. :smile:

File: frontend/app/templates/components/data-component/csv-component.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div class="field">
     <label>Upload CSV</label>
-    <input name="csv" onchange={{action "encodeCSV"}} type="file">
+    <input name="csv" onchange={{action "encodeCSV"}} type="file" accept=".csv">
```
Comment (ParthS007) [inline frontend/app/templates/components/data-component/csv-component.hbs]: @sk9331657 This is already present. Please remove it. :+1:

**Comment by sk9331657** [comment]
@ParthS007  Done Changes.

**Pull request merged**